code/scripting/api/objs/texture.cpp

```diff
@@ -117,6 +117,24 @@ ADE_FUNC(unload, l_Texture, NULL, "Unloads a texture from memory", NULL, NULL)
 	return ADE_RETURN_NIL;
 }
 
+ADE_FUNC(destroyRenderTarget, l_Texture, nullptr, "Destroys a texture's render target. Call this when done drawing to a texture, as it frees up resources.", nullptr, nullptr)
+{
+	texture_h* th;
+
+	if (!ade_get_args(L, "o", l_Texture.GetPtr(&th)))
+		return ADE_RETURN_NIL;
+
+	if (!th->isValid())
+		return ADE_RETURN_NIL;
+
+	if(!bm_is_render_target(th->handle))
+		LuaError(L, "Tried to destroy a render target of a non-renderable texture!");
```

Comment:
`LuaError` can still return normally so you also need to `return` here to ensure that `bm_release_rendertarget` is not called.